### ISSUE TITLE
Catch `ctrl+shift+c`

### ISSUE THREAD

**Issue by Privat33r-dev** [play-with-docker/sdk#59]
Currently there is no way to copy text from the console. Usual combination `ctrl+c` triggers the escape code "^C", which is a valid response, but I don't see a workaround for text copying. `ctrl+shift+c` calls the developer tools in chrome-based browsers.

**Comment by marcosnils** [comment]
> Currently there is no way to copy text from the console. Usual combination `ctrl+c` triggers the escape code "^C", which is a valid response, but I don't see a workaround for text copying. `ctrl+shift+c` calls the developer tools in chrome-based browsers.

have you tried ctrl + inser to copy and shift + insert to paste? that should work

**Comment by Privat33r-dev** [comment]
> > Currently there is no way to copy text from the console. Usual combination `ctrl+c` triggers the escape code "^C", which is a valid response, but I don't see a workaround for text copying. `ctrl+shift+c` calls the developer tools in chrome-based browsers.
> 
> have you tried ctrl + inser to copy and shift + insert to paste? that should work

Thanks, I just found out about that in the upstream's discussion https://github.com/xtermjs/xterm.js/issues/2478

I think that it should be documented since even right mouse button click copying doesn't work by some reason (no option in context menu).

Alternatively, ctrl+shift+c event could be caught.

**Comment by marcosnils** [comment]
> I think that it should be documented since even right mouse button click copying doesn't work by some reason (no option in context menu).

it's documented here: https://github.com/play-with-docker/play-with-docker/?tab=readme-ov-file#how-can-i-use-copy--paste-shortcuts

**Comment by marcosnils** [comment]
> Alternatively, ctrl+shift+c event could be caught.

I'd prefer not to do it since it overrides the browser defaults in some cases. I personally don't like overriding browser default behavior

**Comment by Privat33r-dev** [comment]
> it's documented here: https://github.com/play-with-docker/play-with-docker/?tab=readme-ov-file#how-can-i-use-copy--paste-shortcuts

PWD have tech-savvy userbase, but should it be expected that they will read the repository's README? Maybe the hint can be provided somewhere on the frontend.

## Example
### Original
![image](https://github.com/play-with-docker/sdk/assets/39376984/e7058e9f-8f1a-4494-946f-df1ab218ed53)
### Changed to
![image](https://github.com/play-with-docker/sdk/assets/39376984/daf5da06-6255-47c5-8cac-fe5f105ccbae)

It feels natural, there are other ideas (miniFAQ on server start), but I am not sure about them, this one feels as the easiest solution (though it should be implemented downstream).

**Comment by marcosnils** [comment]
> but I am not sure about them, this one feels as the easiest solution (though it should be implemented downstream).

Yeah.. this one is tricky given that users in training.play-with-docker.com won't be able to see it :pray:

**Comment by Privat33r-dev** [comment]
> > but I am not sure about them, this one feels as the easiest solution (though it should be implemented downstream).
> 
> Yeah.. this one is tricky given that users in training.play-with-docker.com won't be able to see it 🙏

[TryHackMe](https://tryhackme.com/), for example, has a special room that explains how to use rooms. Well, I guess that some general FAQ/help, reference to the button can be added in the machine's footer.

I'd also suggest to do something with footer and header, moving them only on the left part like it's done here (though it's a topic for other issue). And then making a small control element on the bottom, showing timer, machine state and FAQ button + control element to restart machine.

![image](https://github.com/play-with-docker/sdk/assets/39376984/e1c4db27-1ffe-40d9-ab75-249616ac64b7)

**Comment by marcosnils** [comment]
> I'd also suggest to do something with footer and header, moving them only on the left part like it's done here (though it's a topic for other issue). And then making a small control element on the bottom, showing timer, machine state and FAQ button + control element to restart machine.

yes, that'd be awesome. This requires bit of work that unfortunately I don't have the bandwidth to pursue ATM :grimacing:

**Comment by Privat33r-dev** [comment]
> yes, that'd be awesome. This requires bit of work that unfortunately I don't have the bandwidth to pursue ATM 😬

I can do that :)

Btw, what do you think about referral link to the TryHackMe as they have similar rooms, so if people want to continue their education on cybersecurity in a similar way (web rooms), some users can continue with them? I got the idea when I noticed the similarities.
